modules/apps/frontend-js/frontend-js-web/src/main/java/com/liferay/frontend/js/web/internal/JavaScriptPortalWebResources.java

```diff
@@ -65,6 +83,18 @@ protected void setServletContext(ServletContext servletContext) {
 	}
 
 	private Bundle _bundle;
+	private AtomicLong _lastModified;
+	private final ResourceBundleServiceListener _resourceBundleServiceListener =
+		new ResourceBundleServiceListener();
 	private ServletContext _servletContext;
 
+	private class ResourceBundleServiceListener implements ServiceListener {
+
+		@Override
+		public void serviceChanged(ServiceEvent serviceEvent) {
+			_lastModified.incrementAndGet();
```

Comment:
@marianoalvarosaiz I doubt this is right. The getLastModified() use this timestamp to build URLs return to client. It was purely based on the bundle updates, now with this change, the timestamp can "randomly" bump between starts. Meaning the client "bookmarked" url no longer knows it is up to date or not.

How is the resource bundle related to this last modified time? It is not describe in the ticket.

And all the servicelisteners are notified tons of times during startup, due to we have a lot buildin ResourceBundle published. This is going to impact startup time. It must be avoid. You just want to react for runtime deployed resource bundles, not those buildin ones during startup.

Please solve these issues and update the tickets with reproducible steps for each case you are fixing. And upload all test wars you use.